pkg/controllers/provisioning/nodepool_test.go

```diff
@@ -0,0 +1,1577 @@
+/*
```

Comment:
Can we make sure that we didn't duplicate the testing here. All of the NodePool provisioning tests got moved into `suite_test.go` now so we should make sure to move all of the new tests over there and delete this file.